### ISSUE TITLE
Proxy listening to multiple ports

### ISSUE THREAD

**Issue by edemaine** [jwilder/nginx-proxy#921]
I'd like `http://host:80` to forward to one docker, and `http://host:4000` to forward to another docker.  Essentially I want `VIRTUAL_HOST=host:80` for one docker and `VIRTUAL_HOST=host:4000` for another docker.  Is this possible?  (This is different from `VIRTUAL_PORT`, which is specifying the exposed port on the docker.)

Obviously I could do this with two nginx-proxy's.  The challenge is that I actually want to do this with both http and https traffic, using [docker-letsencrypt-nginx-proxy-companion](https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion), with two ports for accessing each docker.  Maybe that'll still be OK with two nginx-proxy's?  Or is there a cleaner way?

**Comment by Laykou** [comment]
I would say, that this PR fixes the issue: https://github.com/jwilder/nginx-proxy/pull/259

It just needs to be merged. If you want to use it now, you can download the [nginx.tmpl](https://raw.githubusercontent.com/MashupMill/nginx-proxy/6d56b1eb35780ecb043c85780a6b63acd2e825cf/nginx.tmpl) file locally and mount it into docker image using volumes

```
sudo docker run --name=nginx-proxy \
  --detach \
  --publish 80:80 \
  --publish 443:443 \
  --restart always \
  --volume /usr/share/nginx/certs:/etc/nginx/certs:ro \
  --volume /etc/nginx/vhost.d \
  --volume /usr/share/nginx/html \
  --volume /var/run/docker.sock:/tmp/docker.sock:ro  \
  --volume /app/nginx-proxy/nginx.tmpl:/app/nginx.tmpl:rw  \
  --label com.github.jrcs.letsencrypt_nginx_proxy_companion.nginx_proxy \
  jwilder/nginx-proxy
```

where `/app/nginx-proxy/nginx.tmpl` is the downloaded tmpl file on my local disc.

**Comment by edemaine** [comment]
Actually #607 looks like exactly what I want.  Thanks for pointing out the large body of available PRs!

**Comment by publicvirtualvoid** [comment]
@Laykou Why does the template need to be rw?
I get this when following your instructions:
```
ERROR: for nginx-proxy  Cannot create container for service nginx-proxy: readdirent: not a directory
ERROR: Encountered errors while bringing up the project.
```

**Comment by Laykou** [comment]
@publicvirtualvoid It does not need to be rw. It was just my limited knowledge about Docker, time and everything. It can be simply `ro` I think :)

**Comment by pmall** [comment]
I'm looking for a solution for the same problem (http://host:80 => one container, http://host:4000 => another container). Sorry but actually the solution mentioned above does not work. I tried both templates and they group the two containers under the same upstram block. @edemaine did you find a solution for this? Thanks.

**Comment by edemaine** [comment]
@pmall No, I didn't solve this, I'm afraid. I was waiting for a release with #607 merged (though perhaps that's not in the cards), and living without it meanwhile.  FWIW, this is what I would like this feature for: https://github.com/edemaine/kadira-compose

> Unfortunately, because of the way letsencrypt-nginx-proxy-companion works, I need ports 80 and 443 to point to kadira-engine instead of kadira-ui. So you need to access the UI via http port 4000.

**Comment by buchdag** [comment]
Implemented in #2678
